src/main/java/com/googlecode/jsonrpc4j/JsonRpcInterceptor.java

```diff
@@ -0,0 +1,70 @@
+package com.googlecode.jsonrpc4j;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+/**
+ * Interceptor can work with parsed full JSON RPC {@link JsonNode} and with parsed target, method and arguments list.
+ * In case you want to work with raw input you can use filters.
+ * In case you want to work with your parsed objects you can use aspect on your exported service.
+ * <p>
+ * Interceptors could throw {@link RuntimeException}, in this case JSON RPC standard error will be shown, except preHandleJson.
+ * In case preHandleJson throws exception, it will generate HTTP code 400 "Bad Request" with empty body. So be careful.
+ * @author pavyazankin
+ * @since 21/09/2017
+ */
+public interface JsonRpcInterceptor {
+
+    /**
+     * Method to intercept raw JSON RPC json objects. Don't throw exceptions here.
+     * In case preHandleJson throws exception, it will be HTTP code 400 "Bad Request" with empty body. So be careful.
```

Comment:
Is it possible for you to handle exceptions from this method? People will end up throwing exceptions in this method; could be confusing for users if the response content changes depending on where the exception is thrown.

Reply:
In progress...

Reply:
It's possible to handle but process of handling will differ from other methods. Problem is that we don't have any request details that need for method  handleError(OutputStream output, Object id, String jsonRpc, AMethodWithItsArgs methodArgs, Throwable e) [375]
We didn't parse jsonRpc request yet. We know its JSON, but what is in it? In this situation I don't even know are there any jsonRpc version, correct method name, arguments. And if I'll try to parse them I could fail with other exception...
Document differences in behaviour in this case is the best choice for us.